lib/autoproj/package_manifest.rb

```diff
@@ -216,5 +222,41 @@ def tag_end(name)
                 @tag_text = nil
             end
         end
+
+        # @api private
+        #
+        # REXML stream parser object used to load the XML contents into a
+        # {PackageManifest} object
+        class RosLoader < Loader
+            DEPEND_TAGS = Set['depend', 'build_depend', 'build_export_depend',
+                              'buildtool_depend', 'buildtool_export_depend',
+                              'exec_depend', 'test_depend', 'run_depend']
+
+            def tag_start(name, attributes)
+                if DEPEND_TAGS.include?(name)
+                    @tag_text = ''
+                elsif TEXT_FIELDS.include?(name) || AUTHOR_FIELDS.include?(name)
+                    @tag_text = ''
+                elsif name == 'tags'
+                    @tag_text = ''
+                else
+                    @tag_text = nil
+                end
+            end
+
+            def tag_end(name)
+                if DEPEND_TAGS.include?(name)
+                    manifest.add_dependency(@tag_text)
```

Comment:
Autoproj does support `test_depend`

Reply:
Yeah, but that relies on the user actually using the `test` sub command. Given that (I am assuming) tests are layed out differently in catkin packages, I don't think that would be a good thing.

Reply:
> Given that (I am assuming) tests are layed out differently in catkin packages, I don't think that would be a good thing.

Can't the ROS people at BIR help you with this ? I have the feeling that building tests and the rest of the package(s) is separated in catkin anyways.

Reply:
> Can't the ROS people at BIR help you with this ?

Not really, unfortunately.

> I have the feeling that building tests and the rest of the package(s) is separated in catkin anyways

Catkin macros' build them automatically if dependencies are met (unless you explicitly disable the tests). I don't there's a good way to make these macros and `autoproj test` play nicely together.

Reply:
(Questions below for my understanding, I'm totally OK with your argument - let's just handle all dependencies the same right now, and refine later)

> **automatically**

Yeah ... that sounds like catkin.

> (unless you explicitly disable the tests).

And you can't do that on a per-package basis ?

Reply:
> And you can't do that on a per-package basis ?

I'm guessing there's probably a cmake definition that allows that but I didn't find it documented anywhere and given that ROS' documentation is quite good, if there's such API it is most likely not considered stable.